cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -170,7 +192,7 @@ func NewSupervisor(logger *zap.Logger, configFile string) (*Supervisor, error) {
 	}
 
 	if connErr := s.waitForOpAMPConnection(); connErr != nil {
-		return nil, fmt.Errorf("failed to connect to the OpAMP server: %w", connErr)
+		logger.Debug("failed to connect to the OpAMP server.", zap.Error(connErr))
```

Comment:
Do we even need `waitForOpAMPConnection` anymore? It looks like we don't do anything special other than potentially wait 10 seconds. It seems like we could just remove this function entirely and have the same result.